apps/web/src/hooks/useTrackThemeChange.ts

```diff
@@ -0,0 +1,21 @@
+import { ColorScheme, useMantineColorScheme } from '@mantine/core';
+import { useLocalThemePreference, useSegment } from '@novu/shared-web';
+import { useEffect } from 'react';
+import { useDebounce } from './useDebounce';
+
+type ThemeChange = { colorScheme: ColorScheme; themeStatus: string };
+
+export default function useTrackThemeChange() {
+  const { themeStatus } = useLocalThemePreference();
+  const { colorScheme } = useMantineColorScheme();
+
+  const segment = useSegment();
+
+  const trackThemeChange = useDebounce<ThemeChange>((args) => {
+    segment.track('Theme is set - [Theme]', args);
+  }, 500);
+
+  useEffect(() => {
+    trackThemeChange({ colorScheme, themeStatus });
+  }, [colorScheme, themeStatus, trackThemeChange]);
+}
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯 suggestion:** Please see my comment for the `useToggleColorTheme` hook in `v2/HeaderNav` and combine that functionality here! This will be great for encapsulating everything related to toggling the theme in one place.‏